xknx/dpt/dpt_1byte_control.py

```diff
@@ -0,0 +1,259 @@
+"""Implementation of Basic KNX DPT B1U3 Values (DPT 3.007/3.008).
+
+Very good source of information for interpretation of the standard is
+https://library.e.abb.com/public/78c74aa86d4648b7b9d918485cd4621a/2CDC500051M0203_ApplicationHB_Lighting_EN.pdf#page=34
+
+There are two separate dimming modes sharing the same DPT class:
+
+ * Stepwise dimming
+   The full brightness range is divided into 2^(stepcode-1) intervals.
+   The value is always rounded to full interval boundary, i.e. 30% +25% = 50%, 50% +25% = 75%, 30% -25% = 25%
+
+ * Start-stop dimming
+   Dimming is started with -/+100% (0x1/0x9) and keeps dimming until a STOP diagram (0x0/0x8) is received.
+
+As the same payload in these cases in interpreted completely different it is reasonable to make separate DPT classes.
+"""
+from enum import Enum
+
+from xknx.exceptions import ConversionError
+
+from .dpt import DPTBase
+
+
+class DPTControl(DPTBase):
+    """Abstraction for KNX B1U3 values (DPT 3.007/3.008)."""
+
+    # APCI (application layer control information)
+    APCI_CONTROLMASK = 0x08
+    APCI_STEPCODEMASK = 0x07
+    APCI_MAX_VALUE = APCI_CONTROLMASK | APCI_STEPCODEMASK
+
+    value_type = "control"
+    unit = ""
+    payload_length = 1
+
+    @classmethod
+    def _encode(cls, control, step_code):
+        """Encode control-bit with step-code."""
+        value = 1 if control > 0 else 0
+        value = (value << 3) | (step_code & cls.APCI_STEPCODEMASK)
+        return value
```

Comment:
I actually don't like this at all, as you put in the intrinsic assumption that control is always either 0 or 1. You added the type hints, so the type checker should detect this, but users of the xknx library might not use type checking in their code. Just imagine I would pass control=3.
It would be possible to use use `(control & cls.APIC_CONTROLMASK) << 3`, but as above this diminishs the abstraction, as the `3` contains redundant information which is already contained in `APIC_CONTROLMASK`. So I'll add the type hints, but would prefer to keep the rest.